iree/turbine/kernel/wave/analysis/index_sequence_analysis.py

```diff
@@ -290,28 +291,30 @@ def populate_mma_source_indices(
             dim, dim_index, node.mma_type
         )
     node.index = combine_indices(node.index, index)
-    return [
-        (
-            get_custom(node.lhs),
-            specialize_index(index, {MMA_LHS: 1, MMA_RHS: 0, MMA_ACC: 0}),
-            node.vector_shapes,
-        ),
-        (
-            get_custom(node.rhs),
-            specialize_index(index, {MMA_LHS: 0, MMA_RHS: 1, MMA_ACC: 0}),
-            node.vector_shapes,
-        ),
-        (
-            get_custom(node.acc),
-            specialize_index(index, {MMA_LHS: 0, MMA_RHS: 0, MMA_ACC: 1}),
-            node.vector_shapes,
-        ),
-        (
-            node,
-            specialize_index(index, {MMA_LHS: 0, MMA_RHS: 0, MMA_ACC: 1}),
-            node.vector_shapes,
-        ),
-    ]
+    lhs_tuple = (
+        get_custom(node.lhs),
+        specialize_index(index, {MMA_LHS: 1, MMA_RHS: 0, MMA_ACC: 0}),
+        node.vector_shapes,
+    )
+    rhs_tuple = (
+        get_custom(node.rhs),
+        specialize_index(index, {MMA_LHS: 0, MMA_RHS: 1, MMA_ACC: 0}),
+        node.vector_shapes,
+    )
+    acc_tuple = (
+        get_custom(node.acc),
+        specialize_index(index, {MMA_LHS: 0, MMA_RHS: 0, MMA_ACC: 1}),
+        node.vector_shapes,
+    )
+    mma_tuple = (
+        node,
+        specialize_index(index, {MMA_LHS: 0, MMA_RHS: 0, MMA_ACC: 1}),
+        node.vector_shapes,
+    )
+    # Remove reduction dim from index of accumulator and mma nodes.
```

Comment:
nice! will this fix Nicolas's issue as well?

Reply:
Possibly, but this was fixing a bug which is that since we are propagating mma/acc to init/users of mma, the index they use should not have the reduction dim in it.

Reply:
nice! :)